### PR TITLE
Relocate profile action button

### LINKED ISSUE #772: Relocate follow button
Let's move the follow button down, underneath the bio to stay out of the way of the user's name. We need all the space we can get.

We can have a small (...) context menu next to follow that opens the meta sheet.

Mockup for general direction:

image

### PR DESCRIPTION
**Pull request by bfollington**
Move profile action button down below bio.

Fixes #772 

- Introduced `ProfileHeaderButtonStyle`

image Screenshot 2023-07-10 at 2 32 21 pm

### CODE REVIEW

File: xcode/Subconscious/Shared/Components/Common/Profile/UserProfileHeaderView.swift [hunk 1]
```diff
@@ -13,6 +13,46 @@ enum UserProfileAction {
...
+            : defaultBackgroundColor
+    }
+    
+    func makeBody(configuration: Configuration) -> some View {
+        configuration.label
+        .bold()
+        .foregroundColor(foregroundColor(configuration))
+        .frame(maxWidth: .infinity, minHeight: AppTheme.minTouchSize)
+        .background(backgroundColor(configuration))
+        .cornerRadius(AppTheme.cornerRadius)
```
Comment (bfollington) [inline xcode/Subconscious/Shared/Components/Common/Profile/UserProfileHeaderView.swift:52]: Also looks good with `cornerRadiusSm`

File: xcode/Subconscious/Shared/Components/Common/Profile/UserProfileHeaderView.swift [hunk 2]
```diff
@@ -13,6 +13,46 @@ enum UserProfileAction {
...
+    private func backgroundColor(_ configuration: Configuration) -> Color {
+        configuration.isPressed
+            ? pressedBackgroundColor
+            : defaultBackgroundColor
+    }
+    
+    func makeBody(configuration: Configuration) -> some View {
+        configuration.label
+        .bold()
+        .foregroundColor(foregroundColor(configuration))
```
Comment (bfollington) [inline xcode/Subconscious/Shared/Components/Common/Profile/UserProfileHeaderView.swift:49]: Maybe this should be a seperate `LabelStyle`? Doesn't actually give us anything other than semantic separation though.

File: xcode/Subconscious/Shared/Components/Common/Profile/UserProfileHeaderView.swift [hunk 3]
```diff
@@ -13,6 +13,50 @@ enum UserProfileAction {
...
 }
 
+enum ProfileHeaderButtonVariant {
+    case primary
+    case secondary
+}
+
+/// A rounded rectanglular row button style designed for the header
+/// This may outgrow its narrow role and the name should be updated if it it used elsewhere
+struct ProfileHeaderButtonStyle: ButtonStyle {
```
Comment (gordonbrander) [inline xcode/Subconscious/Shared/Components/Common/Profile/UserProfileHeaderView.swift:23]: Nit: the ProfileHeaderButtonStyle should probably be in its own file.

File: xcode/Subconscious/Shared/Components/Common/Profile/UserProfileHeaderView.swift (hunk 2 above)
Comment (gordonbrander) [reply xcode/Subconscious/Shared/Components/Common/Profile/UserProfileHeaderView.swift:53]: Wouldn't worry about it

**Review by gordonbrander — APPROVED** [review]
LGTM, one nit.

**Pull request merged**